app/api/products/climatic_summary/router.py

```diff
@@ -0,0 +1,28 @@
+from fastapi import APIRouter, Depends
+from app.api.products.climatic_summary.schema import ClimaticSummaryParams
+from sqlalchemy.orm.session import Session
+from app.api.products.schema import ProductDataParams
+from app.db import get_session
+from app.services.climatic_summary import FailedCreatingClimaticSummary, climatic_summary_create
+from app.utils.product_data import generateProductData
+from app.utils.response import get_error_response, get_success_response
+
+
+router = APIRouter()
+
+
+@router.post("/")
+def create(
+    data_params: ProductDataParams,
+    product_params: ClimaticSummaryParams,
+    db_session: Session = Depends(get_session),
+):
+    data = generateProductData(data_params, db_session)
```

Comment:
This is the shared method for extracting data from the passed parameters

Reply:
Understood. I will try and update `generateProductData()` so that it returns a data frame that can be sent directly to `climatic_summary_create()` (or another service in the `services` folder).